### DISCUSSION TITLE
Exploring the Use of Tree-Sitter for Dependency and Module Analysis in Java Projects

### DISCUSSION THREAD

**Discussion by smith-co** [tree-sitter/tree-sitter#2981] [Q&A] [answered]
I am eager to learn about the practical use of tree-sitter within the Java programming community. Specifically, I'm interested in its application for dependency analysis and module analysis within Java projects.

- **Dependency Graph Construction**: Has anyone successfully used tree-sitter to inspect dependencies within a Java project? I'm interested in its capability to scan a Java project, identify all the internal and external dependencies, and then generate a comprehensive dependency graph. This type of tool could be extremely valuable for Java developers, allowing them to visualize and better comprehend the interrelationships between different components of their project.

- **Module Dependency Graphs**: Furthermore, I am keen to understand if tree-sitter has been employed to construct module dependency graphs in Java projects. This would involve examining a project to identify all its modules, understanding their interdependencies, and then creating a visual representation of these relationships. This could be a significant asset for managing large Java projects with intricate module structures, aiding developers in ensuring the modularity and maintainability of their codebase.

I welcome any experiences, examples, or resources related to the use of tree-sitter for these purposes.

**Comment by sogaiu** [comment, marked as answer]
Haven't used it myself, and not specifically Java-related AFAICT, but may be [tree-sitter-graph](https://github.com/tree-sitter/tree-sitter-graph) could be of interest?
